tests/timeseries/test_serializable_ringbuffer.py

```diff
@@ -0,0 +1,107 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the `SerializableRingBuffer` class."""
+
+from __future__ import annotations
+
+import random
+from datetime import datetime, timedelta
+from typing import Any
+
+import numpy as np
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._serializable_ringbuffer import SerializableRingBuffer
+
+FIVE_MINUTES = timedelta(minutes=5)
+_29_DAYS = 60 * 24 * 29
+ONE_MINUTE = timedelta(minutes=1)
+
+
+def load_dump_test(dumped: SerializableRingBuffer[Any]) -> None:
+    """Test ordered ring buffer."""
+    size = dumped.maxlen
+
+    random.seed(0)
+
+    # Fill with data so we have something to compare
+    # Avoiding .update() because it takes very long for 40k entries
+    for i in range(size):
+        dumped[i] = i
+
+    # But use update a bit so the timestamp and gaps are initialized
+    for i in range(0, size, 100):
+        dumped.update(
+            Sample(datetime.fromtimestamp(200 + i * FIVE_MINUTES.total_seconds()), i)
+        )
+
+    dumped.dump()
+
+    # Load old data
+    # pylint: disable=protected-access
+    loaded = SerializableRingBuffer.load(dumped._path)
+    assert loaded is not None
+
+    np.testing.assert_equal(dumped[:], loaded[:])
+
+    # pylint: disable=protected-access
+    assert dumped._datetime_oldest == loaded._datetime_oldest
+    # pylint: disable=protected-access
+    assert dumped._datetime_newest == loaded._datetime_newest
+    # pylint: disable=protected-access
+    assert len(dumped._gaps) == len(loaded._gaps)
+    # pylint: disable=protected-access
+    assert dumped._gaps == loaded._gaps
+    # pylint: disable=protected-access
+    assert dumped._time_range == loaded._time_range
+    # pylint: disable=protected-access
+    assert dumped._sampling_period == loaded._sampling_period
+    # pylint: disable=protected-access
+    assert dumped._time_index_alignment == loaded._time_index_alignment
+
+
+def test_load_dump_short(tmp_path_factory: Any) -> None:
```

Comment:
Why `Any` and not something more specific?

Reply:
Because it was faster and is just a test, not production code, so I didn't invest the time to find the right import for the typehint..

Reply:
Ah, yeah, those magical arguments passed by `pytest` are really convenient but the fact that you don't need to import anything to use them makes it quite hard to figure out the type.